### PR TITLE
Document releasing procedure

### PR DESCRIPTION
**Pull request by p12tic**
This PR adds a short document that documents the releasing procedure.

### CODE REVIEW

File: RELEASING.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+Creating a release
+==================
+
+This document is documentation intended for Buildbot maintainers. It documents the release
```
Comment (seankelly) [inline RELEASING.rst]: Buildbot style is one sentence per line

File: RELEASING.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    make release VERSION=x.y.z
+
+This will create the required tags, make documentation, copy it to bbdocs repo and push everything.
+
+Step 4: Wait for CircleCi to create release tarballs
+----------------------------------------------------
+
+The push of tags created during step 3 will activate CircleCi configuration that generates tarballs
+and uploads them to Github. This is a good time to draft a release on the Github UI
```
Comment (seankelly) [inline RELEASING.rst]: Elsewhere you have "GitHub" so we should be consistent with casing everywhere and change this to "GitHub" too.

File: RELEASING.rst (hunk 1 above)
Comment (p12tic) [reply RELEASING.rst]: Argh, I keep forgetting that it's not only for internal docs :-) Fixed now.

File: RELEASING.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+This step requires GitHub Hub tool to be installed and authorized to GitHub (https://github.com/github/hub).
+Additionally you have to have access to GPG key that is used to sign the releases.
+Finally, you have to be added as a maintainer to all Buildbot PyPi projects.
+
+To complete the release just run the following:
+
+    make finishrelease
+
+The above will download the releases from GitHub and upload them using twine.
+If you get bytes-related error after entering Pypi password, you'll need to upgrade Twine.
```
Comment (tardyp) [inline RELEASING.rst]: I think you forgot the announcement part.

Write a blog post with the list of highlights of the release. A short community update is a plus to help monotony of release notes.
git shortlog also is important to recognise our contributors.

Then annoucement on the three mailing list, with link to the blog post.

Comment (p12tic) [reply RELEASING.rst]: Fixed. I guess it's too late to do this for 1.8.2 and 2.3.1, isn't it?

**Pull request merged**